src/main/java/org/opensearch/neuralsearch/ml/MLCommonsClientAccessor.java

```diff
@@ -236,6 +238,18 @@ private <T extends Number> List<List<T>> buildVectorFromResponse(MLOutput mlOutp
         return resultMaps;
     }
 
+    private String buildQueryResultFromResponseOfOutput(MLOutput mlOutput) {
+        final ModelTensorOutput modelTensorOutput = (ModelTensorOutput) mlOutput;
+        final List<ModelTensors> tensorOutputList = modelTensorOutput.getMlModelOutputs();
+        if (CollectionUtils.isEmpty(tensorOutputList) || CollectionUtils.isEmpty(tensorOutputList.get(0).getMlModelTensors())) {
+            throw new IllegalStateException(
+                "Empty model result produced. Expected at least [1] tensor output and [1] model tensor, but got [0]"
+            );
+        }
+        List<ModelTensor> tensorList = tensorOutputList.get(0).getMlModelTensors();
+        return tensorList.get(0).getResult();
```

Comment:
Any reason to only use the first element? For other inference calls we iterate and add tensors from the whole list because the format of the model tensors (collected in the first element versus from all elements) depends on how we make the model call. 

E.g. for inference map results used for sparse encoding processors, a local model call will have all inference results in the first ModelTensors object, but for a batch remote call, the results are distributed across multiple ModelTensors, requiring us to iterate through all. See also https://github.com/opensearch-project/neural-search/blob/fa5b809b20d5b2b4d1c5ff7f6a4557845302fc0f/src/main/java/org/opensearch/neuralsearch/ml/MLCommonsClientAccessor.java#L230-L235 for reference